app/services/check_cmc_ohclv_service.rb

```diff
@@ -101,16 +103,38 @@ def check_coins
 
         # Check results.
         # This should only be one row since we're only checking one coin at a time (and it's grouped).
-        result.each do |row|
-          check_volume = ranking.present? && ranking < 100
-          has_volume = row["to"].to_f > 0
-          has_results = row["count"] > 0
-
-          # There should be at least one entry and volume should be non-zero if ranking < 100
-          if !has_results || (check_volume && !has_volume) then
-            @failed_coins << coin_key
-            next
-          end
+        row = result.first
+
+        check_volume = ranking.present? && ranking < 100
+        has_volume = row["to"].to_f > 0
+        has_results = row["count"] > 0
+
+        # There should be at least one entry and volume should be non-zero if ranking < 100
+        if !has_results || (check_volume && !has_volume) then
+          @failed_coins << coin_key
+          @failed_cached_coins << coin_key # Assume failed since no db result
+          next
+        end
+
+        # Double-check that we've cached the latest results
+        latest_cached_price_data = nil
+        if @granularity == 'daily'
+          latest_cached_price_data = coin.prices_data.reverse!.first # prices_data is fetched as ASC
+        elsif @granularity == 'hourly'
+          latest_cached_price_data = coin.hourly_prices_data.reverse!.first # hourly_prices_data is fetched as ASC
+        else
+          raise "Could not find cached price data for selected granularity: #{@granularity}"
+        end
+
+        if latest_cached_price_data.blank?
+          @failed_cached_coins << coin_key
+          next
+        end
+
+        latest_cached_timestamp = DateTime.parse(latest_cached_price_data["time"])
+        db_timestamp = DateTime.parse(row["time"])
+        if latest_cached_timestamp != db_timestamp then
```

Comment:
Do we need to parse the time strings prior to comparison? Legit question, I'm not sure, just checking.

Reply:
The cached version is formatted slightly different so a direct string comparison was failing.